bundles/org.openhab.core/src/main/java/org/eclipse/smarthome/core/items/Metadata.java

```diff
@@ -25,15 +25,17 @@
  * This is a data class for storing meta-data for a given item and namespace.
  * It is the entity used for within the {@link MetadataRegistry}.
  *
- * @author Kai Kreuzer - Initial contribution and API
- *
+ * @author Kai Kreuzer - Initial contribution
  */
 @NonNullByDefault
 public final class Metadata implements Identifiable<MetadataKey> {
 
-    private final MetadataKey key;
-    private final String value;
-    private final Map<String, Object> configuration;
+    private MetadataKey key;
+    private String value;
+    private Map<String, Object> configuration;
+
+    Metadata() {
```

Comment:
Don't remove `final`.
Add the comment that is (should be) used on all the other such constructors, too.

Reply:
I get you. Immutable classes are preferable. Is gson able to modify final fields?

Reply:
Gson uses reflection to set final fields (via `.setAccessible(true)`),